src/include/processor/operator/profile.h

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include "processor/operator/physical_operator.h"
+
+namespace kuzu {
+namespace processor {
+
+struct ProfileSharedState {
+    std::unique_ptr<PhysicalPlan> plan;
+
+    explicit ProfileSharedState(std::unique_ptr<PhysicalPlan> plan);
+};
+
+struct ProfileLocalState {
+    common::ValueVector* outputVector;
```

Comment:
Our convention is to not put Vector into local state. Put vector as a field of operator